magicgui/function_gui.py

```diff
@@ -0,0 +1,291 @@
+        """Set the result name of this MagicGui widget."""
+        self._result_name = value
+
+    def Gui(self, show=False):
+        """Create a widget instance [DEPRECATED]."""
+        warnings.warn(
+            "\n\nCreating a widget instance with `my_function.Gui()` is deprecated,\n"
+            "the magicgui decorator now returns a widget instance directly, so you\n"
+            "should simply use the function itself as a magicgui widget, or call\n"
+            "`my_function.show(run=True)` to run the application.\n"
+            "In a future version, the `Gui` attribute will be removed.\n",
+            FutureWarning,
+        )
+        if show:
+            self.show()
+        return self
+
+
+# ==================   magicgui decorator   ===================================
+
+F = TypeVar("F", bound=Callable[..., Any])
+
+
+@overload
+def magicgui(function: F, **k) -> FunctionGui:  # noqa: D103
+    ...
+
+
+@overload
+def magicgui(function=None, **k) -> Callable[[F], FunctionGui]:  # noqa: D103
+    ...
+
+
+def magicgui(
+    function: Optional[F] = None,
+    *,
+    orientation: str = "horizontal",
+    labels: bool = True,
+    call_button: Union[bool, str] = False,
+    auto_call: bool = False,
+    result_widget: bool = False,
+    app: AppRef = None,
+    **param_options: dict,
+):
+    """Create a FunctionGui class for ``function`` and add it as an attribute ``Gui``.
+
+    Parameters
+    ----------
+    function : Callable, optional
+        The function to decorate.  Optional to allow bare decorator with optional
+        arguments. by default None
+    orientation : str, optional
+        The type of layout to use. Must be one of {'horizontal', 'vertical',
+        'form', 'grid'}, by default "horizontal"
+    labels : bool
+        Whether labels are shown in the widget. by default True
+    call_button : bool or str, optional
+        If True, create an additional button that calls the original function when
+        clicked.  If a ``str``, set the button text. by default False
+    auto_call : bool, optional
+        If True, changing any parameter in either the GUI or the widget attributes
+        will call the original function with the current settings. by default False
+    result_widget : bool, optional
+        Whether to display a LineEdit widget the output of the function when called,
+        by default False
+    app : magicgui.Application or str, optional
+        A backend to use, by default None (use the default backend.)
+
+    **param_options : dict of dict
+        Any additional keyword arguments will be used as parameter-specific options.
+        Keywords MUST match the name of one of the arguments in the function
+        signature, and the value MUST be a dict.
+
+    Returns
+    -------
+    Callable
+        The original function is returned with a new attribute ``Gui``.  Gui is a
+        subclass of MagicGui that, when instantiated, will create a widget representing
```

Comment:
I see a few mentions of MagicGui in the codebase. Is that FunctionGui? Or do you refer to the actual package?

Reply:
thank you!  that's vestigial...  will update.